document-generator-core/src/main/java/uk/gov/companieshouse/document/generator/core/document/DocumentGenerator.java

```diff
@@ -0,0 +1,69 @@
+package uk.gov.companieshouse.document.generator.core.document;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import uk.gov.companieshouse.document.generator.core.avro.AvroDeserializer;
+import uk.gov.companieshouse.document.generator.core.document.models.RenderSubmittedDataDocument;
+import uk.gov.companieshouse.document.generator.core.kafka.ConsumerGroupHandler;
+import uk.gov.companieshouse.document.generator.interfaces.DocumentInfoService;
+import uk.gov.companieshouse.document.generator.interfaces.model.DocumentInfo;
+import uk.gov.companieshouse.kafka.consumer.CHKafkaConsumerGroup;
+import uk.gov.companieshouse.kafka.message.Message;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class DocumentGenerator implements Runnable {
+
+    private static final List<String> CONSUMER_TOPICS = Arrays.asList("render-submitted-data-document");
+
+    private static final String GROUP_NAME = "document-generator";
+    
+    private CHKafkaConsumerGroup documentGeneratorConsumerGroup;
+
+    private DocumentInfoService documentInfoService;
+
+    private AvroDeserializer<RenderSubmittedDataDocument> avroDeserializer;
+
+    @Autowired
+    public DocumentGenerator(DocumentInfoService documentInfoService, ConsumerGroupHandler consumerGroupHandler,
+                             AvroDeserializer<RenderSubmittedDataDocument> avroDeserializer) {
+
+        this.documentInfoService = documentInfoService;
+        this.avroDeserializer = avroDeserializer;
+
+        documentGeneratorConsumerGroup =
+                consumerGroupHandler.getConsumerGroup(CONSUMER_TOPICS, GROUP_NAME);
+    }
+
+    @Override
+    public void run() {
+
+        try {
+            pollAndGenerate();
+        } catch (Exception e) {
+
+        }
+    }
+
+    /**
+     * Poll kafka messages and obtain documentInfo data
+     */
+    private void pollAndGenerate() {
+
+        for (Message message : documentGeneratorConsumerGroup.consume()) {
+
+            RenderSubmittedDataDocument renderSubmittedDataDocument;
+
+            try {
+                renderSubmittedDataDocument = avroDeserializer
+                        .deserialize(message, RenderSubmittedDataDocument.getClassSchema());
+
+                DocumentInfo documentInfo = documentInfoService.getDocumentInfo();
+
+                documentGeneratorConsumerGroup.commit();
+            } catch (Exception e) {
```

Comment:
This needs logging to be able to diagnose errors. Can you catch the specific errors to write logs appropriate to the error which has occurred please?

Reply:
I have a task to implement all logging, which will be using the structured logging development work that Guru undertook.

Reply:
Code in Develop should be releasable. (see [GitFlow at CH](https://companieshouse.atlassian.net/wiki/spaces/DEVOPS/pages/78151945/GitFlow+at+CH#GitFlowatCH-developDevelopBranch)) which means all exceptions should be handled and logged appropriately (see [Java review standards](https://github.com/companieshouse/styleguides/blob/master/java_review.md)). Other developers could try to integrate with your code or take over one of your tasks at any time and will be frustrated if they have problems and nothing is getting logged with exceptions ignored.

Until Guru's code is ready can you add an import for the structured-logger please.

Reply:
added Log.error in catch